src/date/expand-pattern.js

```diff
@@ -48,25 +171,38 @@ return function( options, cldr ) {
 			if ( !result ) {
 				timeSkeleton = skeleton.split( /[^hHKkmsSAzZOvVXx]/ ).slice( -1 )[ 0 ];
 				dateSkeleton = skeleton.split( /[^GyYuUrQqMLlwWdDFgEec]/ )[ 0 ];
-				if ( /(MMMM|LLLL).*[Ec]/.test( dateSkeleton ) ) {
-					type = "full";
-				} else if ( /MMMM/g.test( dateSkeleton ) ) {
-					type = "long";
-				} else if ( /MMM/g.test( dateSkeleton ) || /LLL/g.test( dateSkeleton ) ) {
-					type = "medium";
+				dateTimeSkeleton = getBestMatchPattern(
+					"dates/calendars/gregorian/dateTimeFormats/availableFormats",
+					skeleton
+				);
+				if ( dateTimeSkeleton ) {
+					result = dateTimeSkeleton;
 				} else {
-					type = "short";
-				}
-				result = combineDateTime( type,
-					cldr.main([
+					dateSkeleton = getBestMatchPattern(
 						"dates/calendars/gregorian/dateTimeFormats/availableFormats",
 						dateSkeleton
-					]),
-					cldr.main([
+					);
+					timeSkeleton = getBestMatchPattern(
 						"dates/calendars/gregorian/dateTimeFormats/availableFormats",
 						timeSkeleton
-					])
-				);
+					);
+
+					if ( /(MMMM|LLLL).*[Ec]/.test( dateSkeleton ) ) {
+						type = "full";
+					} else if ( /MMMM/g.test( dateSkeleton ) ) {
+						type = "long";
+					} else if ( /MMM/g.test( dateSkeleton ) || /LLL/g.test( dateSkeleton ) ) {
+						type = "medium";
+					} else {
+						type = "short";
+					}
+
+					if ( dateSkeleton && timeSkeleton ) {
+						result = combineDateTime( type, dateSkeleton, timeSkeleton );
+					} else {
+						result = dateSkeleton || timeSkeleton;
+					}
+				}
```

Comment:
In general, the whole block above is getting too indented. Let's change it. Currently, we have something like this:

```js
tryDirectMap()
if (!directMap) {
  tryToMatchIt()
  if (!match) {
    tryToMatchItsIndividualParts()
  }
}
```

Let's re-arrange it so we have:


```js
tryDirectMap()
if (directMap) {
  break;
}

tryToMatchIt()
if (match) {
  break;
}

tryToMatchItsIndividualParts()
```